services/QuillLMS/app/workers/copy_concept_results_to_responses_worker.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+class CopyConceptResultsToResponsesWorker
+  include Sidekiq::Worker
+  sidekiq_options queue: SidekiqQueue::MIGRATION
+
+  BATCH_SIZE=100000
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def perform(start, finish)
+    directions_cache = {}
+    instructions_cache = {}
+    previous_feedbacks_cache = {}
+    prompts_cache = {}
+    question_types_cache = {}
+
+    Response.bulk_insert(ignore: true) do |worker|
+      ConceptResult.where(id: start..finish).each do |concept_result|
```

Comment:
This will load 100,000 records into memory, but only working on one at a time. We should use [find_each](https://apidock.com/rails/ActiveRecord/Batches/find_each) to find this in smaller batches (even something like 10,000 records). Or you could adjust the batch size.